src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -657,7 +657,13 @@ public CitizenData getCitizenData()
     {
         if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY)
         {
-            return (T) new InvWrapper(getCitizenData().getInventory());
+            final CitizenData data = getCitizenData();
+            if (data == null)
+            {
+                return null;
```

Comment:
This is by API definition not the correct solution for this problem. We can not return null when we return true on hasCapability.

Reply:
Agreed, please change :smiley:

Reply:
then `hasCapability` should also be corrected to return false?